exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillPushLimitToScanRule.java

```diff
@@ -55,18 +62,21 @@ public void onMatch(RelOptRuleCall call) {
     }
   };
 
-  public static DrillPushLimitToScanRule LIMIT_ON_PROJECT =
-      new DrillPushLimitToScanRule(
-          RelOptHelper.some(DrillLimitRel.class, RelOptHelper.some(
-              DrillProjectRel.class, RelOptHelper.any(DrillScanRel.class))),
-          "DrillPushLimitToScanRule_LimitOnProject") {
+  public static DrillPushLimitToScanRule LIMIT_ON_PROJECT = new DrillPushLimitToScanRule(
+      RelOptHelper.some(DrillLimitRel.class, RelOptHelper.any(DrillProjectRel.class)), "DrillPushLimitToScanRule_LimitOnProject") {
     @Override
     public boolean matches(RelOptRuleCall call) {
       DrillLimitRel limitRel = call.rel(0);
-      DrillScanRel scanRel = call.rel(2);
-      // For now only applies to Parquet. And pushdown only apply limit but not offset,
+      DrillProjectRel projectRel = call.rel(1);
+      // pushdown only apply limit but not offset,
       // so if getFetch() return null no need to run this rule.
-      if (scanRel.getGroupScan().supportsLimitPushdown() && (limitRel.getFetch() != null)) {
```

Comment:
I can understand that this check was removed because this matches() method no longer is checking for DrillScanRel, but does that mean that no one is checking the GroupScan for supportsLimitPushdown() ?

Reply:
We still have the LIMIT_ON_SCAN rule which does that check. This rule is changed from LIMIT_PROJECT_SCAN to LIMIT_PROJECT. The LIMIT_SCAN along with the LIMIT_PROJECT would work as the LIMIT_PROJECT_SCAN.

Reply:
One implication of this is suppose the underlying Scan does not support Limit pushdown, you could end up with a plan `Scan->Limit->Project->Limit`  where the Limit above the Scan is redundant (assume that there is no FLATTEN in this query).  Can this be avoided ?

Reply:
Without a FLATTEN, the LIMIT would be fully pushed past the PROJECT i.e. we would not have a LIMIT on top of the project.

Reply:
Ok, yeah in that case we are not generating a redundant limit.